compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -5764,17 +5764,28 @@ public void visit(BLangXMLSequenceLiteral bLangXMLSequenceLiteral, AnalyzerData
             data.resultType = symTable.xmlTextType;
             return;
         }
+        
+        if (expType.tag == TypeTags.ANY) {
+            data.resultType = symTable.anyType;
+            return;
+        }
+
+        if (expType.tag == TypeTags.ANYDATA) {
+            data.resultType = symTable.anydataType;
+            return;
+        }
+        
         // Disallow unions with 'xml:T (singleton) items
-         for (BType item : ((BUnionType) expType).getMemberTypes()) {
-             item = Types.getReferredType(item);
-             if (item.tag != TypeTags.XML_TEXT && item.tag != TypeTags.XML) {
-                 dlog.error(bLangXMLSequenceLiteral.pos, DiagnosticErrorCode.INCOMPATIBLE_TYPES,
-                         expType, symTable.xmlType);
-                 data.resultType = symTable.semanticError;
-                 return;
-             }
-         }
-        data.resultType = symTable.xmlType;
+        for (BType item : ((BUnionType) expType).getMemberTypes()) {
```

Comment:
Can we guarantee the `expType` is always union at this point, can't that be `XML_TEXT` (based on the if check in L:5737)

Reply:
Based on the checks in L:5754 (`XML`, `noType`), L:5763 (`XML_TEXT`), L:5768 (`ANY`), L:5773 (`ANYDATA`), the other types make use of an early return. Therefore, we can guarantee the `expType` is always `UNION` at this point, right?